clients/src/test/java/org/apache/kafka/common/record/MemoryRecordsTest.java

```diff
@@ -149,10 +144,29 @@ public void testIterator(Args args) {
             new SimpleRecord(6L, (byte[]) null, null)
         };
 
-        for (SimpleRecord record : records)
-            builder.append(record);
+        final MemoryRecords memoryRecords;
+        try (var builder = new MemoryRecordsBuilder(
+                buffer,
+                magic,
+                compression,
+                TimestampType.CREATE_TIME,
+                firstOffset,
+                logAppendTime,
+                pid,
+                epoch,
+                firstSequence,
+                false,
+                false,
+                partitionLeaderEpoch, buffer.limit()
```

Comment:
nit: as like other params.

```suggestion
                partitionLeaderEpoch,
                buffer.limit()
```

Reply:
Thanks. Fixed.